grammars/ruby.cson

```diff
@@ -479,6 +479,8 @@
           |
           if|else|elsif|then|do|end|unless|while|until|or|and
         )
+        |
+        $
```

Comment:
This wouldn't be needed if we test what goes before `/` instead. This is a modified version of the begin pattern from TextMate's bundle:

```cson
(?:
  ^
  |
  (?<=
    [(\\[{~&=|;:,>?]|
    [\\s;]when\\s|
    [\\s;]if\\s|
    [\\s;]elsif\\s|
    [\\s;]else\\s|
    [\\s;]while\\s|
    [\\s;]unless\\s|
    [\\s;]then\\s|
    [\\s;]or\\s|
    [\\s;]and\\s|
    [\\s;]not\\s|
    [\\s.]match\\s|
    [\\s.]gsub\\s|
    [\\s.]gsub!\\s|
    [\\s.]sub\\s|
    [\\s.]sub!\\s|
    [\\s.]slice\\s|
    [\\s.]slice!\\s|
    [\\s.]index\\s|
    [\\s.]scan\\s|
    [\\s.]split\\s|
    [\\s.]partition\\s|
    [\\s.]rpartition\\s|
    [\\s;]assert_match\\s
  )
  |
  (?<=
    ^when\\s|
    ^if\\s|
    ^elsif\\s|
    ^while\\s|
    ^unless\\s
  )
)
\\s*((/))(?![*+?])
```

Replacing the begin pattern with this fixes issue 144 and 82, but it doesn't work when a method that accepts a regexp as its first arg (a method that's not listed in the pattern) is called without putting parentheses around the args as in `user_defined_method /foo/`. Considering the Ruby community encourages using parentheses around args (with some exceptions), I think this could be accepted compared to if you have to escape `/` in a character class.

What do you think? @50Wliu